src/Compilers/CSharp/Portable/Symbols/UnsupportedMetadataTypeSymbol.cs

```diff
@@ -13,6 +14,9 @@ internal UnsupportedMetadataTypeSymbol(BadImageFormatException mrEx = null)
             _mrEx = mrEx;
         }
 
+        protected override NamedTypeSymbol WithTupleDataCore(TupleExtraData newData)
+            => throw ExceptionUtilities.Unreachable;
```

Comment:
I fear that this could crash the compiler when there is an issue with the imported tuple type.  Perhaps the implementation should just be `=> this`